docs/reference/install-kubernetes.md

```diff
@@ -263,15 +239,91 @@ INFO[0000] Experimental features enabled
 ...
 ```
 
+#### Customizing Inspektor Gadget
+
+##### Hook Mode
+
+Inspektor Gadget needs to detect when containers are started and stopped.
+The different supported modes can be set by using the `hook-mode` option:
+
+- `auto`(default): Inspektor Gadget will try to find the best option based on
+  the system it is running on.
+- `crio`: Use the [CRIO
+  hooks](https://github.com/containers/podman/blob/v3.4.4/pkg/hooks/docs/oci-hooks.5.md)
+  support. Inspektor Gadget installs the required hooks in
+  `/etc/containers/oci/hooks.d`, be sure that path is part of the `hooks_dir`
+  option on
+  [crio.conf](https://github.com/cri-o/cri-o/blob/v1.20.0/docs/crio.conf.5.md#crioruntime-table).
+  If `hooks_dir` is not declared at all, that path is considered by default.
+- `podinformer`: Use a Kubernetes controller to get information about new pods.
+  This option is racy and the first events produced by a container could be
+  lost. This mode is selected when `auto` is used and the above modes are not
+  available.
+- `nri`: Use the [Node Resource Interface](https://github.com/containerd/nri).
+  It requires containerd v1.5 and it's not considered when `auto` is used.
+- `fanotify+ebpf`:  Uses the Linux
+  [fanotify](https://man7.org/linux/man-pages/man7/fanotify.7.html) API and an
+  eBPF module. It works with both runc and crun. It works regardless of the
+  pid namespace configuration.
+
+In order to set the hook mode start by creating daemon configuration file, for example `daemon-config.yaml`:
+
+```bash
+cat <<EOF > daemon-config.yaml
+hook-mode: fanotify+ebpf
+EOF
+```
+
+Then, deploy Inspektor Gadget with the following command:
+
+```
+$ kubectl gadget deploy --daemon-config=daemon-config.yaml
+```
+
+Following is a sample `daemon-config.yaml` file to showcase the different options:
+
+```yaml
+containerd-socketpath: /run/containerd/containerd.sock
+crio-socketpath: /run/crio/crio.sock
+daemon-log-level: info
+docker-socketpath: /run/docker.sock
+events-buffer-length: 16384
+fallback-pod-informer: true
+gadget-namespace: gadget
+hook-mode: auto
+operator:
+  oci:
+    allowed-gadgets: []
+    disallow-pulling: false
+    insecure-registries: []
+    public-keys:
+      - |
+        -----BEGIN PUBLIC KEY-----
+        MFkwEwYHKoZIzj0CAQYIKoZIzj0DAQcDQgAEoDOC0gYSxZTopenGmX3ZFvQ1DSfh
+        Ir4EKRt5jC+mXaJ7c7J+oREskYMn/SfZdRHNSOjLTZUMDm60zpXGhkFecg==
+        -----END PUBLIC KEY-----
+    verify-image: true
+  otel-metrics:
+    otel-metrics-listen: false
+    otel-metrics-listen-address: 0.0.0.0:2224
+podman-socketpath: /run/podman/podman.sock
+```
+
+###### Other Deploy Options
+
+Please check the following documents to learn more about different options:
+- [Restricting the Gadgets that can be run](./restricting-gadgets.mdx)
+- [Using Insecure Registries](./insecure-registries.mdx)
+- [Verifying Gadget Images](./verify-assets.mdx#verify-image-based-gadgets)
+
```

Comment:
Should we also update those places to use the config file?

Reply:
I think we missed https://inspektor-gadget.io/docs/main/reference/restricting-gadgets (that's the only one I checked, so that's why I thought they all were missing)